src/heartbeat/utils.js

```diff
@@ -114,26 +117,24 @@ function setupRepeater(generator) {
   debug('setupRepeater %O', sanitized);
   if (commonUtils.isBulk(generator)) {
     debug('Generator %s is bulk', generator.name);
-    repeater.createGeneratorRepeater(generator, collect, handleCollectResponse);
+    repeater.createGeneratorRepeater(generator, collectBulk, handleCollectResponse);
   } else {
-    // FIXME bulk is false
-    generator.subjects.forEach((s) => {
-      const _g = JSON.parse(JSON.stringify(generator));
-      _g.subjects = [s];
-      repeater.createGeneratorRepeater(_g, collect, handleCollectResponse);
-    });
+    debug('Generator %s is non-bulk', generator.name);
+    repeater.createGeneratorRepeater(generator, collectNonBulk, handleCollectResponseNonBulk);
   }
 } // setupRepeater
 
 /**
  * Update queue for sample generator (if found), or create a new one.
  *
  * @param  {String} qName - Queue name
+ * @param  {String} token - The Authorization token to use.
+ * @param  {Number} intervalSecs - time before generator will have new data available
```

Comment:
in the context of this function, "intervalSecs" is not the right name for this. How if the queue using this variable? Can you give it a better name which describes how it's being used?

Reply:
i went with 'flushFunctionCutoff'. thoughts?